ofono/manager.go

```diff
@@ -46,18 +46,24 @@ func NewModemManager(conn *dbus.Connection) *ModemManager {
 }
 
 func (mm *ModemManager) Init() error {
-	modemAddedSignal, err := connectToSignal(mm.conn, "/", OFONO_MANAGER_INTERFACE, "ModemAdded")
+	//Use a different connection for the modem signals to avoid go-dbus blocking issues
+	conn, err := dbus.Connect(dbus.SystemBus)
+	if err != nil {
+		log.Fatal("Connection error in ModemManager: ", err)
```

Comment:
Maybe just return err? I'm almost sure whatever handles Init will log.Fatal the return value (feel free to double check)